### ISSUE TITLE
Application Error caused by Run Bot (manual bot run from dashboard)

### ISSUE THREAD

**Issue by Stoner19** [Steem-FOSSbot/steem-fossbot-voter#86]
I'm able to edit the config, edit algorithm, and test the algorithm, but when I tell it to "Run Bot" it hangs for a little while then I get:
```
Application error
An error occurred in the application and your page could not be served. If you are the application owner, check your logs for details.
```

**Comment by thrize** [comment]
This is something I'm aware of but put on the back burner. It is caused because the process is taking longer than the page load timeout time. It was okay when the bot was made originally because there was less activity but now it takes longer, particularly due to increasing ```MAX_POST_TO_READ``` (see #70).

I will have to make this asynchronous in some way. I'll add this as priority as I feel new users will encounter this an be discouraged.

Thanks for reminding me.

**Comment by Stoner19** [comment]
Thanks for the info. Yeah, I discovered yesterday that its due to the 30 second timeout limit in place at Heroku. If I see a solution I'll be sure to report back.

**Comment by thrize** [comment]
The solution would be to perform the bot run asynchronously and check in on the result later, loading a page while it's working so we see something.

The usual way to run the bot is with a scheduled job, see [here](https://github.com/Steem-FOSSbot/steem-fossbot-voter/blob/master/docs/installation.md#important-configure-bot-to-run-periodically) for details.

**Comment by Skytec** [comment]
Aww. I did discover this project today in the docker so I see no possibility on how to run the bot seperately from the interface..

**Comment by Stoner19** [comment]
@Skytec Its not the interface nor the code, its entirely the 30 second http default timeout of Heroku

**Comment by Skytec** [comment]
Ah, okay, since I deployed it in Docker. Does your Docker deployment depends on Heroku in any way?

**Comment by thrize** [comment]
No the Docker deployment doesn't depend on Heroku.

If you start up the Docker deployment using the friendly helper scripts you will be asked what what frequency you want to run the bot, i.e. how often.

<img width="554" alt="docker" src="https://user-images.githubusercontent.com/25388336/27960374-facf7b3c-6322-11e7-994f-f59e391b24ac.png">

If this is not working please open a bug ticket on [the docker project](https://github.com/Steem-FOSSbot/voter-docker/issues)

**Comment by thrize** [comment]
Anyway, I'll implement a fix for this.

**Comment by thrize** [comment]
Fixed for v0.2.10

**Comment by kende** [comment]
Only see v0.2.9 available in this repo. Where can I find v0.2.10?

**Comment by antoniotrento** [comment]
I've this problem too, and then if I set the scheduled job, the heroku app refresh it every hour but if I go to `herokuapp.com/stats` page to see if something is done, the page don't show me stats but the message: 

> 'It looks like this is a fresh install of Voter. Please generate some stats by using it and then come back here to see the results in detail.'

So I was not able to figure out if the bot works or not.

PS: I set the algorithm with basic settings to see how it works, and I tried the 'Test Algorithm' area it seems that the algorithm finds post with the specifications described in the algorithm

**Comment by thrize** [comment]
Hmm, I'll get on to this when I resume work

**Comment by thrize** [comment]
This is on my radar guys, working on this stuff this week

**Comment by SpiryMedia** [comment]
The same issue encountered. 
Can't wait for the fix. I am trying to schedule the process to see if it starts this way.

**Comment by thrize** [comment]
Fixed by #64
